js/data/buffer/triangle_element_buffer.js

```diff
@@ -1,27 +1,24 @@
 'use strict';
 
 var util = require('../../util/util');
-var Buffer = require('./buffer');
-
-module.exports = TriangleElementBuffer;
-
-function TriangleElementBuffer(buffer) {
-    Buffer.call(this, buffer);
+var Buffer2 = require('../buffer2');
+
+function TriangleElementBuffer(options) {
+    Buffer2.call(this, options || {
+        type: Buffer2.BufferType.ELEMENT,
+        attributes: {
+            verticies: {
+                components: 3,
+                type: Buffer2.AttributeType.UNSIGNED_SHORT
+            }
+        }
```

Comment:
Yes, the attributes could move to the prototype, but ideally they wouldn't move to the prototype. The raison-d'etre of `Buffer2` is that it has a configurable buffer layout (vs `Buffer1` which had a hardcoded buffer layout). Layout is intended to be passed as a constructor argument at runtime. This code represents an intermediary step towards that goal.